main/src/ca/uwaterloo/flix/runtime/channel/GlobalCounter.java

```diff
@@ -0,0 +1,10 @@
+package flix.runtime;
+import java.util.concurrent.atomic.AtomicInteger;
+
+public class GlobalCounter {
+    private static final AtomicInteger gloablCounter = new AtomicInteger();
```

Comment:
I think this needs to be a long. An Int32 holds 4 billion channels. You could imagine for a long running program - or a program that create many channels - that this bound could be exceeded.

Reply:
Also there seems to be a spelling error in the counter name.